sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/mdesupport/AzureKeyVaultKeyWrapProvider.java

```diff
@@ -0,0 +1,101 @@
+import com.azure.cosmos.implementation.HttpConstants;
+import com.azure.cosmos.implementation.ImplementationBridgeHelpers;
+import com.microsoft.data.encryption.AzureKeyVaultKeyStoreProvider.AzureKeyVaultKeyStoreProvider;
+import com.microsoft.data.encryption.cryptography.MicrosoftDataEncryptionException;
+
+/**
+ * Provides an implementation for an Azure key store provider. A DEK encrypted with a key store provider
+ * should be decryptable by this provider and vice versa.
+ * <p>
+ * Envelope Format for the encrypted data encryption key: version + keyPathLength + ciphertextLength + keyPath +
+ * ciphertext + signature.
+ * <p>
+ * version: A single byte indicating the format version.
+ * <p>
+ * keyPathLength: Length of the keyPath.
+ * <p>
+ * ciphertextLength: ciphertext length.
+ * <p>
+ * keyPath: keyPath used to encrypt the data encryption key. This is only used for troubleshooting purposes,
+ * and is not verified during decryption.
+ * <p>
+ * ciphertext: Encrypted data encryption key.
+ * <p>
+ * signature: Signature of the entire byte array. Signature is validated before decrypting the data encryption key.
+ */
+public class AzureKeyVaultKeyWrapProvider extends EncryptionKeyWrapProvider {
+    private AzureKeyVaultKeyStoreProvider azureKeyVaultKeyStoreProvider;
+    private final static ImplementationBridgeHelpers.CosmosExceptionHelper.CosmosExceptionAccessor cosmosExceptionAccessor =
+        ImplementationBridgeHelpers.CosmosExceptionHelper.getCosmosExceptionAccessor();
+
+    public AzureKeyVaultKeyWrapProvider(TokenCredential tokenCredential) {
+        try {
+            this.azureKeyVaultKeyStoreProvider = new AzureKeyVaultKeyStoreProvider(tokenCredential);
+        } catch (MicrosoftDataEncryptionException ex) {
+            throw cosmosExceptionAccessor.createCosmosException(HttpConstants.StatusCodes.INTERNAL_SERVER_ERROR, ex);
+        }
+    }
+
+    /**
+     * Getter for name of the key vault provider.
+     *
+     * @return provider name
+     */
+    @Override
+    public String getProviderName() {
+        return this.azureKeyVaultKeyStoreProvider.getProviderName();
+    }
+
+    /**
+     * Decrypts an encrypted DEK with the RSA encryption algorithm using the asymmetric key specified by the key path
+     *
+     * @param encryptionKeyId                 - Complete path of an asymmetric key in Azure Key Vault
+     * @param cosmosKeyEncryptionKeyAlgorithm - Asymmetric Key Encryption Algorithm
+     * @param encryptedKey                    - Encrypted Data Encryption Key
+     * @return Plain text data encryption key
+     */
+    @Override
+    public byte[] unwrapKeyAsync(String encryptionKeyId, String cosmosKeyEncryptionKeyAlgorithm, byte[] encryptedKey) {
+        try {
+            if (!KeyEncryptionKeyAlgorithm.RSA_OAEP.equals(cosmosKeyEncryptionKeyAlgorithm)) {
+                throw new IllegalArgumentException("The specified KeyEncryptionAlgorithm is not supported. Please " +
```

Comment:
should we put the algorithm name in the message?

Reply:
This is if something change in the AAP which support more algorith , cosmos will have only RSA_OAEP as of now , same as .NET https://github.com/Azure/azure-cosmos-dotnet-v3/blob/5c01524de44ea0104223f6d494edeb98ac30a44c/Microsoft.Azure.Cosmos.Encryption/src/MdeSupport/AzureKeyVaultKeyWrapProvider.cs#L52